scripts/gitlab/audit.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e # fail on any error
+set -u # treat unset variables as error
+
+rustup default stable
+
+git submodule update --init --recursive
```

Comment:
we don't really need submodules here

```
cargo +stable install cargo-audit
cargo +stable audit
```
would be enough

Btw, is this job required or allowed to fail? Looks like you've removed all `allow_failure`s.

Reply:
Also, for `cargo install`s it'd nice to specify `CARGO_TARGET_DIR` to enable caching (see https://github.com/jtgeibel/cargo/blob/a81b1437de95920618119272c7a92bab9974ee30/src/etc/man/cargo-install.1#L45-L47).

Reply:
I was under the impression we cache all jobs? see line 16 in .gitlab.yml

Reply:
it is at the end of the pipeline and can happily fail :)

Reply:
> I was under the impression we cache all jobs? see line 16 in .gitlab.yml

Yes, but it looks like we specify `CARGO_TARGET_DIR` only in [wasm-tests](https://github.com/paritytech/wasm-tests/blob/0edbf860ff7ed4b6b6336097ba44836e8c6482dd/build-rust-test.sh#L9)?

Reply:
we default to ./target everywhere and that should be cached per job name, just double-checked.